File: 2026POTD/MARCH/03-03-2026/1545.py
class Solution:
    def findKthBit(self, n: int, k: int) -> str:

        invert = 0

        l = 1

        for _ in range(1, n):
            l = l * 2 + 1

        init = "0"

        while l > 1:

            half = l // 2
            if half + 1 == k:
                init = "1"
                break
            elif k > half + 1:
                invert += 1
                k = l + 1 - k

            l = half

        if invert % 2 != 0:
            if init == "0":
                init = "1"
            else:
                init = "0"

        return init

File: 2026POTD/MARCH/03-03-2026/1545.java
class Solution {
    public String reverse(String str){
        StringBuilder sb=new StringBuilder(str);
        return sb.reverse().toString();
    }
    public String invert(String s){
        StringBuilder str=new StringBuilder();
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='0')str.append('1');
            else str.append('0');
        }
        return str.toString();
    }
    public String construct(int n){
        if(n==0)return "0";
        return construct(n-1)+"1"+reverse(invert(construct(n-1)));
    }
    public char findKthBit(int n, int k) {
        String s= construct(n);
        return s.charAt(k-1);
    }
}